Sources/MapboxCoreNavigation/Router.swift

```diff
@@ -268,47 +266,54 @@ extension InternalRouter where Self: Router {
                     routeOptions = options
                 }
                 
-                self?.indexedRouteResponse = .init(routeResponse: response, routeIndex: 0)
-                self?.setRoute(route, proactive: true, routeOptions: routeOptions)
+                // Prefer the most optimal route (the first one) over the route that matched the original choice.
+                self.indexedRouteResponse = .init(routeResponse: response, routeIndex: 0)
+                
+                // If the upcoming maneuver in the new route is the same as the current upcoming maneuver, don’t announce it.
+                // FIXME: There must be a better way to skip a redundant initial instruction than to assume the old route’s spoken instruction index is appliable to the new route.
+                let spokenInstructionIndex = self.routeProgress.currentLegProgress.currentStepProgress.spokenInstructionIndex
+                self.routeProgress = RouteProgress(route: route, options: routeOptions ?? self.routeProgress.routeOptions, legIndex: 0, spokenInstructionIndex: spokenInstructionIndex)
```

Comment:
While inlining `setRoute(_:proactive:routeOptions:)` here, I restored an old comment from before #1986 that explained why we try to persist the spoken instruction index in the new route progress. Apparently we consider it confusing to hear the “Head northwest on Main Street” instruction out of the blue without the rerouting chime.

This bit of gymnastics dates back to when we originally implemented proactive rerouting in #346. At the time, there was no concept of an instruction index, only an “alert level” that increased as the user neared the maneuver point. Maintaining the alert level made sense as a way to suppress that initial instruction, since the alert level would’ve been guaranteed to be nonzero at any time beyond the beginning of the old step.

We migrated to server-generated instructions in #614, eliminating the concept of alert levels in favor of instruction indices. 903e0207722f889abffe191dde52fdc8799331e8 migrated this code to persist the instruction index, but that introduced a potential problem: unlike alert levels, there’s theoretically an arbitrary number of instruction indices that could vary from one step to the next, or between two steps belonging to two different routes.

I don’t know for sure that this persistence is causing any problems, so I’ve retained the behavior for now, but it needs a closer look. There’s got to be a better way to suppress the initial departure instruction, if we still think that’s necessary.